subprojects/core/src/main/java/org/gradle/internal/service/scopes/GradleScopeServices.java

```diff
@@ -152,13 +153,14 @@ CommandLineTaskParser createCommandLineTaskParser(OptionReader optionReader, Tas
     }
 
     BuildWorkExecutor createBuildExecuter(StyledTextOutputFactory textOutputFactory, IncludedBuildControllers includedBuildControllers, BuildOperationExecutor buildOperationExecutor) {
-        return new BuildOperationFiringBuildWorkerExecutor(
-            new IncludedBuildLifecycleBuildWorkExecutor(
-                new DefaultBuildWorkExecutor(
-                    asList(new DryRunBuildExecutionAction(textOutputFactory),
-                        new SelectedTaskExecutionAction())),
-                includedBuildControllers),
-            buildOperationExecutor);
+        return new DeprecateUndefinedBuildWorkExecutor(
```

Comment:
I would keep the build operation as the outer most executor unless this breaks something.  That way the deprecation will be associated with the outermost build operation